components/event-receiver/org.wso2.carbon.event.input.adapter.core/src/main/java/org/wso2/carbon/event/input/adapter/core/internal/InputAdapterRuntime.java

```diff
@@ -84,9 +84,13 @@ public void start() {
     public void destroy() {
         if (inputEventAdapter != null) {
             try {
-                inputEventAdapter.disconnect();
-            } finally {
-                inputEventAdapter.destroy();
+                try {
```

Comment:
It would make more readable if you could move outer try block inside the finally block as it only catches throwable occur at destroy() method.